src/session_impl.cpp

```diff
@@ -6070,14 +6070,14 @@ namespace {
 		m_alerts.emplace_alert<dht_live_nodes_alert>(nid, nodes);
 	}
 
-	void session_impl::dht_sample_infohashes(udp::endpoint const& ep, sha1_hash const& target)
+	void session_impl::dht_sample_infohashes(sha1_hash const& nid, udp::endpoint const& ep, sha1_hash const& target)
 	{
 		if (!m_dht) return;
-		m_dht->sample_infohashes(ep, target, [this, ep](time_duration const interval
+		m_dht->sample_infohashes(ep, target, [this, nid, ep](time_duration const interval
```

Comment:
I would have expected this callback to get passed in the node ID instead of just capturing *some* node ID passed in by the called